Copy operation state field from Application status

chetan-rns opened the pull request:
#### Description:
- Add a new field `operationState` to the status of GitOpsDeployment CR
- Add a new field to store the operation state in the ApplicationState DB table
- Copy the operation state from Application to GitOpsDeployment
- Compress/Decompress the operation state before accessing it from the table
- Add unit tests and e2e tests

#### Link to JIRA Story (if applicable):

https://issues.redhat.com/browse/GITOPSRVCE-619

chetan-rns commented:
/retest

jparsai commented:
Hey @chetan-rns there are conflicts in PR, please resolve them, so that it is good for review.

jparsai commented:
/lgtm

jgwest reviewed (changes requested):
Looks great, thanks @chetan-rns, and thanks for reviewing @jparsai! Some minor unit test changes that occurred to me during reviewing, then good to go.

chetan-rns commented on backend-shared/db/types.go:
```diff
@@ -303,6 +303,8 @@ type ApplicationState struct {
 
 	Resources []byte `pg:"resources"`
 
+	OperationState []byte `pg:"operation_state"`
```
Added the new field in db-migration test

chetan-rns commented:
@jgwest @jparsai I've added the missing tests. Also, I moved some common functions (compress/decompress) to backend-shared to avoid duplication. PTAL.

jgwest reviewed (approved):
Looks great, thanks again @chetan-rns!
